components/InputBox/index.tsx

```diff
@@ -83,6 +285,236 @@ const InputBox = ({
     };
   }, []);
 
+  // function calls a confirm alert which will further call onSend function onConfirm.
+  const sendDmRequest = () => {
+    showDMSentAlert();
+  };
+
+  const showDMSentAlert = () => {
+    setDMSentAlertModalVisible(true);
+  };
+
+  const hideDMSentAlert = () => {
+    setDMSentAlertModalVisible(false);
+  };
+
+  // function checks if we have access of storage in Android.
+  async function requestStoragePermission() {
+    if (Platform.OS === 'android') {
+      let OSVersion = Platform.constants['Release'];
+
+      if (Number(OSVersion) < 13) {
+        try {
+          const granted = await PermissionsAndroid.request(
+            PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
+            {
+              title: 'Storage Permission',
+              message: 'App needs permission to access your storage',
+              buttonNeutral: 'Ask Me Later',
+              buttonNegative: 'Cancel',
+              buttonPositive: 'OK',
+            },
+          );
+          if (granted === PermissionsAndroid.RESULTS.GRANTED) {
```

Comment:
Create a PermissionUtil class or function and move this code to that place, what is the use of calling this permission from InputBox, it should from chatroom or media picker screen.